### PR TITLE
Location reminder in task

### PR DESCRIPTION
**Pull request by sydneyhauke**
Addresses #227.

For now it justs adds a (nullable) Location field in the Task class.

I would like your input on what I could further add.

### CODE REVIEW

File: app/src/main/java/com/github/steroidteam/todolist/model/todo/Task.java [hunk 1]
```diff
@@ -39,6 +42,15 @@ public String getBody() {
         return this.body;
     }
 
+    public void setLocationReminder(Location location) {
+        Objects.requireNonNull(location);
```
Comment (YagoGG) [inline app/src/main/java/com/github/steroidteam/todolist/model/todo/Task.java:46]: If we add this condition (which I think is good, just like we do with due dates), we need another method to remove the location reminder. See `Task.removeDueDate` for reference.

**Review by YagoGG — CHANGES_REQUESTED** [review]
Hey, @sydneyhauke! Thanks a bunch for this.

I am missing a method (see inline comment), as well as the corresponding modifications to the `JSONSerializerTest`. See #249 for context, and #212 as an example.

Cheers!

**Comment by sydneyhauke** [comment]
Ok, so I decided to use this already existing branch for issue #247. Now it it possible to add a location reminder in a task by pinpointing on a map. The location remains persistent in the database.

**Comment by sydneyhauke** [comment]
Ok, so I decided to use this already existing branch for issue #247. Now it it possible to add a location reminder in a task by pinpointing on a map. The location remains persistent in the database.

**Review by theo-damiani — APPROVED** [review]
Nice PR, very good work!

For me, we can merge it, maybe just remove the TodoReposiroty and ItemViewModel file.

**Review by YagoGG — APPROVED** [review]
Thanks a lot for the work, @sydneyhauke!

ship it!

**Pull request merged**